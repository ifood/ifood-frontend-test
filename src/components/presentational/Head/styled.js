import styled from "styled-components";
import media from "styled-media-query";

import * as T from "constants/styles/typography";
import * as C from "constants/styles/colors";
import { SPACINGS } from "constants/styles/spacings";

export const Head = styled.header`
  padding: ${SPACINGS.lg} 0;
  background-color: ${C.GREY_LIGHT};
`;

export const HeadContent = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  ${media.lessThan("large")`
    flex-direction: column;
  `}
`;

export const HeadTitle = styled.h1`
  color: ${C.PRIMARY_COLOR};
  font-weight: 700;
  font-size: ${T.SIZES.lg};

  ${media.lessThan("large")`
    margin-bottom: ${SPACINGS.xs}
  `}
`;

export const HeadTagline = styled.p`
  color: ${C.GREY_DARK};
  font-size: ${T.SIZES.sm};

  span {
    color: ${C.PRIMARY_COLOR};
    font-weight: 700;
  }
`;
