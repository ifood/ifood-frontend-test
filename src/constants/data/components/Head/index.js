export const headData = {
  title: "Spotifood",
  tagline: "Where music meets <span>iFood</span>",
};
