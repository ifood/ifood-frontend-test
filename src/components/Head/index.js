import React from "react";
import PropTypes from "prop-types";
import parser from "html-react-parser";

import Wrapper from "components/Wrapper";

import { headData } from "constants/data/components/Head";

import * as S from "./styled";

const Head = ({ title, tagline }) => {
  return (
    <S.Head>
      <Wrapper>
        <S.HeadContent>
          <S.HeadTitle>{title}</S.HeadTitle>
          <S.HeadTagline>{parser(tagline)}</S.HeadTagline>
        </S.HeadContent>
      </Wrapper>
    </S.Head>
  );
};

Head.defaultProps = {
  title: headData.title,
  tagline: headData.tagline,
};

Head.propTypes = {
  title: PropTypes.string,
  tagline: PropTypes.string,
};

export default Head;
