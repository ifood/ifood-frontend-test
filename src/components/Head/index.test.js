import React from "react";
import { render } from "@testing-library/react";

import Head from "./index";

test("<Head/> renders correcly", () => {
  const props = {
    title: "Spotifood",
    tagline: "Where music meets iFood",
  };

  const component = <Head {...props} />;
  const { asFragment } = render(component);
  expect(asFragment(component)).toMatchSnapshot();
});
