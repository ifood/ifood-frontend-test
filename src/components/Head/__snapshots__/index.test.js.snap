// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Head/> renders correcly 1`] = `
<DocumentFragment>
  <header
    class="sc-gtssRu Usrn"
  >
    <div
      class="sc-bdnylx cNZMtW"
    >
      <div
        class="sc-dlnjPT gjuKFn"
      >
        <h1
          class="sc-hKFyIo gIJbbs"
        >
          Spotifood
        </h1>
        <p
          class="sc-eCApGN hKpdGl"
        >
          Where music meets iFood
        </p>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
